src/applications/static-pages/mhv-signin-cta/test/MhvSigninCallToAction.unit.spec.jsx

```diff
@@ -0,0 +1,134 @@
+      expect(result.serviceName).to.eql(CSP_IDS.ID_ME);
+      expect(result.userIsVerified).to.eql(true);
+    });
+  });
+
+  describe('render widget', () => {
+    const mockStore = configureStore([]);
+    const serviceDescription = 'order supplies';
+    const linkText = 'order medical supplies';
+
+    /**
+     * Creates an HTML Collection with a div and a link.
+     * @returns the HTML Collection
+     */
+    const noAlertContent = () => {
+      // The link
+      const linkEl = document.createElement('a');
+      linkEl.appendChild(document.createTextNode(linkText));
+      linkEl.href = '/health-care/order-medical-supplies/';
+      linkEl.title = linkText;
+
+      // The widget content div
+      const widgetContent = document.createElement('div');
+      widgetContent.className = 'static-widget-content';
+      widgetContent.appendChild(linkEl);
+      return widgetContent;
+    };
+
+    it('unanthenticated user', async () => {
+      const { queryByRole } = render(
+        <Provider store={mockStore()}>
+          <MhvSigninCallToAction
+            serviceDescription={serviceDescription}
+            userIsLoggedIn={false}
+            noAlertContent={noAlertContent()}
+          />
+        </Provider>,
+      );
+      expect(
+        queryByRole('heading', { name: RegExp(unanthenticatedHeadingPrefix) }),
+      ).to.exist;
+      expect(queryByRole('heading', { name: RegExp(unverfiedHeadingPrefix) }))
+        .to.be.null;
+      expect(queryByRole('link', { name: RegExp(linkText) })).to.be.null;
+    });
+
+    it('unverified user', async () => {
+      const { queryByRole } = render(
+        <Provider store={mockStore()}>
+          <MhvSigninCallToAction
+            serviceDescription={serviceDescription}
+            userIsLoggedIn
+            userIsVerified={false}
+            serviceName={CSP_IDS.ID_ME}
+            noAlertContent={noAlertContent()}
+          />
+        </Provider>,
+      );
+      expect(
+        queryByRole('heading', { name: RegExp(unanthenticatedHeadingPrefix) }),
+      ).to.be.null;
+      expect(queryByRole('heading', { name: RegExp(unverfiedHeadingPrefix) }))
+        .to.exist;
+      expect(queryByRole('link', { name: RegExp(linkText) })).to.be.null;
+    });
+
+    it('verified user', async () => {
+      const { queryByRole } = render(
+        <Provider store={mockStore()}>
+          <MhvSigninCallToAction
+            serviceDescription={serviceDescription}
+            userIsLoggedIn
+            userIsVerified
+            serviceName={CSP_IDS.ID_ME}
+            noAlertContent={noAlertContent()}
+          />
+        </Provider>,
+      );
+      expect(
+        queryByRole('heading', { name: RegExp(unanthenticatedHeadingPrefix) }),
+      ).to.be.null;
+      expect(queryByRole('heading', { name: RegExp(unverfiedHeadingPrefix) }))
+        .to.be.null;
+      expect(queryByRole('link', { name: RegExp(linkText) })).to.exist;
```

Comment:
Suggestion: Since your unit specs for these components already have assertions for headings, links, and analytics you could make a more-general, static assertion about the component being rendered, such as with testIds. No change necessary.

Reply:
I see. I'll take a look.

Reply:
This makes sense and I changed the code. I did test for the service description text to make sure it is passed down to the components.